donate/payment.py

```diff
@@ -0,0 +1,158 @@
+    payload['udf2'] = ac.academic_code
+    purpose = payee_obj_new.purpose
+    if  purpose == 'cdcontent':
+        payload['udf5'] = 'cdcontent'
+    else:
+        payload['udf5'] = TrainingEvents.objects.get(id=purpose).event_name
+    return payload
+    
+def make_hdfc_session_request(payee_obj_new, headers, payload):
+    try:
+        response = requests.post(settings.HDFC_API_URL, json=payload, headers=headers)
+        response_data = response.json()
+    except requests.exceptions.RequestException as e:
+        # messages.add_message(request, messages.ERROR, "An error occurred. Please try later")
+        # return render(request, template, context=context)
+        return None
+    if response.status_code == 200:
+        payment_links = response_data.get("payment_links", {})
+        payment_link = payment_links.get("web")
+        if not payment_link:
+            # messages.error(request, "Payment link is missing. Please contact support.")
+            # return render(request, template, context=context)
+            return None
+        transaction = save_ilw_hdfc_session_data(response_data)
+        payee_obj_new.transaction = transaction
+        payee_obj_new.save()
+        return payment_link
+    else:
+        transaction = save_ilw_hdfc_session_data(response_data, payee_obj_new.amount)
+        payee_obj_new.transaction = transaction
+        payee_obj_new.save()
+        return None
+    return None
+
+@csrf_exempt
+def check_ilw_payment_status(request, order_id):
+    """
+    API endpoint for frontend to check payment status.
+    This triggers backend polling.
+    """
+    payee = Payee.objects.get(transaction__order_id=order_id)
+    amount = payee.transaction.amount
+    email = payee.email
+    result = poll_payment_status(order_id, email, amount)
+    return JsonResponse(result)
+
+
+def poll_payment_status(order_id, email, sub_amount):
+    """
+    Polls the HDFC API every 15 seconds to check the payment status.
+    Stops polling when:
+      - The status is "charged" or "failed"
+      - The max attempts is reached
+    """
+    HDFC_STATUS_URL = f"{settings.ORDER_STATUS_URL}{order_id}"
+    headers = get_request_headers(email)
+    max_attempts = settings.HDFC_POLL_MAX_RETRIES 
+    wait_time = settings.HDFC_POLL_INTERVAL # Wait before polling again
+    attempt = 0
+    while attempt < max_attempts:
+        try:
+            response = requests.get(HDFC_STATUS_URL, headers=headers)
+            data = response.json()
+            if response.status_code == 200 and "status" in data:
+                order_status = data.get('status', '')
+                amount = data.get('amount', '')
+                order_id = data.get('order_id', '')
+                try:
+                    amount_decimal = Decimal(str(amount))
+                except InvalidOperation:
+                    save_ilw_hdfc_error_data(order_id, data, msg="Amount error")
+                    return {"status": "ERROR"}
```

Comment:
The try should be inside the if order_status == 'CHARGED' case. when status is say PENDING, the amount may not be available.